### ISSUE TITLE
code makes an outdated assumption about the primary branch

### ISSUE THREAD

**Issue by davidalpert** [remotemobprogramming/mob#160]
When I follow the instructions for [How to Contribue](https://github.com/remotemobprogramming/mob#how-to-contribute) virtually all the tests are failing.

It appears that is because I have configured my local git installation to use `main` as the default branch name but the tests and code have a different assumption.

This seems related to #82 but not quite the same.

I propose adding a new configuration option called `c.DefaultBranch`/`MOB_DEFAULT_BRANCH` (or perhaps `c.PrimaryBranch`/`MOB_PRIMARY_BRANCH`?) and replacing all hard-coded references to the current assumption.

**Comment by simonharrer** [comment]
The master mob-session stuff is legacy and only kept for backwards compatibility reasons. The new format is basebranch and mob/basebranch. So with main it would be mob/main. I want to keep it that way. That’s why I am against changing any production code.

Regardless, the tests should work with any initialDefault branch git setting. So you uncovered a testing issue here. We probably only need to fix the createtestbed script so it will always use master regardless of any initialDefault setting. We can later change that to main but that would be a large change and doesn’t provide a lot of value. I would first try to fix that testing bug before discussing larger changes.

**Comment by davidalpert** [comment]
if I update PR #82 to default to `master` the rest of refactoring that setting to be configurable should be backwards compatible, no?

**Comment by davidalpert** [comment]
also, my initialDefault branch git setting doesn't match all the repos I work with 🤔

**Comment by davidalpert** [comment]
this is interesting.

I reverted the changes to the tests, removed `/tmp/mob`, updated the createtestbed script to use this command:
```
(cd "${remotedir}" && git --bare init --initial-branch=master)
```

but the tests still failed with the wrong branch errors:
```
SET WORKING DIR TO /tmp/mob/local
======================
mob_test.go:981:

        exp: "on branch master"

        got: "on branch main"

--- FAIL: TestStatusNotMobProgramming (0.51s)
```

when I ran `./createtestbed` here's what I found:
```
~: cat /tmp/mob/remote/HEAD 
ref: refs/heads/master
~: cat /tmp/mob/local/.git/HEAD 
ref: refs/heads/main
~
```
suggesting that while the current branch was set as expected in the remote repo that was not preserved when the repo was cloned; perhaps because the remote repo had no commits so the git client picked it's configured default branch on clone?

**Comment by davidalpert** [comment]
okay, have to explicitly create the `master` branch in the `local` (the working copy of the bare remote) and push it back to remote to properly set the active branch in the remote.

looks like git clone sets the local HEAD using an algorithm which looks at references pointing to the current commit that was cloned. so when the createtestbed script created the bare remote repo, then cloned the empty remote repo into the local folder, git didn't find any commits or any refs so set HEAD to it's default branch name.

explicitly creating the `master` branch in the `local` repo and pushing it back to the remote seems to work.

**Comment by davidalpert** [comment]
closed #162 and replaced it with #164 which contains just the testbed fix

**Comment by davidalpert** [comment]
I will close this issue now, as I have not yet seen any issues with using the mob tool in a repo where the default branch is `main`; I will reopen if I encounter a concrete use case which requires revisiting #162
